src/dist/replication/lib/replica_split.cpp

```diff
@@ -606,6 +606,183 @@ void replica::parent_check_sync_point_commit(decree sync_point) // on primary pa
+            b,
+            get_ballot());
+        return;
+    }
+
+    partition_configuration child_config = _primary_states.membership;
+    child_config.ballot++;
+    child_config.last_committed_decree = 0;
+    child_config.last_drops.clear();
+    child_config.pid.set_partition_index(_app_info.partition_count +
+                                         get_gpid().get_partition_index());
+
+    register_child_request request;
+    request.app = _app_info;
+    request.child_config = child_config;
+    request.parent_config = _primary_states.membership;
+    request.primary_address = _stub->_primary_address;
+
+    // reject client request
+    update_local_configuration_with_no_ballot_change(partition_status::PS_INACTIVE);
+    set_inactive_state_transient(true);
+    _partition_version = -1;
+
+    parent_send_register_request(request);
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::parent_send_register_request(
+    const register_child_request &request) // on primary parent
+{
+    dassert_replica(status() == partition_status::PS_INACTIVE,
+                    "current status should be inactive, wrong status = {}",
+                    enum_to_string(status()));
+
+    ddebug_replica(
+        "send register child({}) request to meta_server, current ballot = {}, child ballot = {}",
+        request.child_config.pid,
+        request.parent_config.ballot,
+        request.child_config.ballot);
+
+    rpc_address meta_address(_stub->_failure_detector->get_servers());
+    std::unique_ptr<register_child_request> req = make_unique<register_child_request>(request);
+    register_child_rpc rpc(std::move(req), RPC_CM_REGISTER_CHILD_REPLICA);
+    _primary_states.register_child_task =
+        rpc.call(meta_address,
+                 tracker(),
+                 [this, rpc](error_code ec) mutable {
+                     on_register_child_on_meta_reply(ec, rpc.request(), rpc.response());
+                 },
+                 _split_states.parent_gpid.thread_hash());
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::on_register_child_on_meta_reply(
+    dsn::error_code ec,
+    const register_child_request &request,
+    const register_child_response &response) // on primary parent
+{
+    FAIL_POINT_INJECT_F("replica_on_register_child_on_meta_reply", [](dsn::string_view) {});
+
+    _checker.only_one_thread_access();
+
+    // primary parent is under reconfiguration, whose status should be PS_INACTIVE
+    if (partition_status::PS_INACTIVE != status() || _stub->is_connected() == false) {
+        dwarn_replica("status wrong or stub is not connected, status = {}",
+                      enum_to_string(status()));
+        _primary_states.register_child_task = nullptr;
+        // TODO(heyuchen): TBD - clear other split tasks in primary context
+        return;
+    }
+
+    if (ec != ERR_OK || (ec == ERR_OK && response.err != ERR_OK)) {
```

Comment:
```    
    dsn::error_code err = ec == ERR_OK ? response.err : ec;
    if (err != ERR_OK) {
```

Reply:
For short-circuit evaluation, you can use `if (ec != ERR_OK || response.err != ERR_OK)` instead, of course @levy5307 's suggestion would be better, you can reuse `err` in the following logging.